src/pages/AccountDeletion.js

```diff
@@ -0,0 +1,84 @@
+import { useEffect, useState, useContext } from "react";
+import { UserContext } from "../contexts/UserContext";
+import InfoPage from "./info_page_shared/InfoPage";
+import Header from "./info_page_shared/Header";
+import Heading from "./info_page_shared/Heading";
+import Main from "./info_page_shared/Main";
+import Footer from "./info_page_shared/Footer";
+import LoadingAnimation from "../components/LoadingAnimation";
+import redirect from "../utils/routing/routing";
+import SessionStorage from "../assorted/SessionStorage";
+import { APP_DOMAIN } from "../appConstants";
+
+const TIME_BEFORE_REDIRECT = 5000;
+
+export default function AccountDeletion({ api }) {
+  const user = useContext(UserContext);
+  const [headingMsg, setHeadingMsg] = useState("We are deleting your account");
+  const [errorOccurred, setErrorOccurred] = useState();
+  const [isAccountDeleted, setIsAccountDeleted] = useState();
+  useEffect(() => {
+    console.log(user);
+    if (SessionStorage.hasUserConfirmationForAccountDeletion()) {
+      setErrorOccurred(false);
```

Comment:
I'm getting an warning in the IDE: `Argument type boolean is not assignable to parameter type (prevState: undefined) => undefine` for both errorOccurred and isAccountDeleted. I think this is because they are not initialized. Can't we simply initialize them with `false`?

Reply:
Yes! We can set it to false.

The reason why I stopped initializing the React states is in order to use the loading component, if you set it to false it's much harder to know when you actually have updated the API. In this case, we can just render the component and respond after one of them is toggled, but in other cases I don't see an easy way without leaving it undefinied. 

To be honest, I don't understand why that warning would appear, as you can assign whatever Type to a React state, as we never have to specify types when we create a react state? Or am I wrong?

Reply:
I know: It's really funny and I don't fully understand why this is. 

One thing is to simply disable that warning. 

The other would be to from now on, not use boolean variables, but a locally declared enum that tracks the possible states of the respective variables, e.g. the above would become something like: `deletionStatus` would have one of the values `UNDEFINED`, `ERRORED`, `OK`. Does this make sense?

Reply:
I kind of like this! What is your feeling about it?

Reply:
I also like it! 

I am not sure if we need the React states anymore of `isAccountDeleted` and `ErrorOcurred`, as they are not used anymore - so if we like this I would remove those to avoid confusion.

Reply:
yes, let's!

Reply:
We should be ready then!